src/dodal/devices/panda_fast_grid_scan.py

```diff
@@ -0,0 +1,288 @@
+        return GridAxis(values["z2_start"], values["z_step_size"], values["z_steps"])
+
+    def is_valid(self, limits: XYZLimitBundle) -> bool:
+        """
+        Validates scan parameters
+
+        :param limits: The motor limits against which to validate
+                       the parameters
+        :return: True if the scan is valid
+        """
+        x_in_limits = limits.x.is_within(self.x_axis.start) and limits.x.is_within(
+            self.x_axis.end
+        )
+        y_in_limits = limits.y.is_within(self.y_axis.start) and limits.y.is_within(
+            self.y_axis.end
+        )
+
+        first_grid_in_limits = (
+            x_in_limits and y_in_limits and limits.z.is_within(self.z1_start)
+        )
+
+        z_in_limits = limits.z.is_within(self.z_axis.start) and limits.z.is_within(
+            self.z_axis.end
+        )
+
+        second_grid_in_limits = (
+            x_in_limits and z_in_limits and limits.y.is_within(self.y2_start)
+        )
+
+        return first_grid_in_limits and second_grid_in_limits
+
+    def get_num_images(self):
+        return self.x_steps * self.y_steps + self.x_steps * self.z_steps
+
+    @property
+    def is_3d_grid_scan(self):
+        return self.z_steps > 0
+
+    def grid_position_to_motor_position(self, grid_position: ndarray) -> ndarray:
+        """Converts a grid position, given as steps in the x, y, z grid,
+        to a real motor position.
+
+        :param grid_position: The x, y, z position in grid steps
+        :return: The motor position this corresponds to.
+        :raises: IndexError if the desired position is outside the grid."""
+        for position, axis in zip(
+            grid_position, [self.x_axis, self.y_axis, self.z_axis]
+        ):
+            if not axis.is_within(position):
+                raise IndexError(f"{grid_position} is outside the bounds of the grid")
+
+        return np.array(
+            [
+                self.x_axis.steps_to_motor_position(grid_position[0]),
+                self.y_axis.steps_to_motor_position(grid_position[1]),
+                self.z_axis.steps_to_motor_position(grid_position[2]),
+            ]
+        )
+
+
+class GridScanCompleteStatus(DeviceStatus):
+    """
+    A Status for the grid scan completion
+    Progress bar functionality has been removed for now in the panda fast grid scan
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.start_ts = time.time()
+
+        self.device.status.subscribe(self._running_changed)
+
+        self._name = self.device.name
+
+        self._target_count = self.device.y_steps.get()
+
+    def _running_changed(self, value=None, old_value=None, **kwargs):
+        if (old_value == 1) and (value == 0):
+            self.set_finished()
+            self.clean_up()
+
+    def clean_up(self):
+        self.device.status.clear_sub(self._running_changed)
```

Comment:
Could: This is exactly the same as the old one, can we just put it in a common place?

Reply:
Re-added as the progress bar functionality is removed in the Panda version